packages/flutter/lib/src/services/autofill.dart

```diff
@@ -0,0 +1,287 @@
+  /// on iOS, [.AUTOFILL_HINT_PHONE_NUMBER](https://developer.android.com/reference/androidx/autofill/HintConstants#AUTOFILL_HINT_PHONE_NUMBER)
+  /// on Android, and [tel](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete)
+  /// on web.
+  static const String telephoneNumber = 'telephoneNumber';
+
+  /// The input field expects the first line of a street address.
+  ///
+  /// Translates to [UITextContentType.streetAddressLine1](https://developer.apple.com/documentation/uikit/uitextcontenttype/1649663-streetaddressline1),
+  /// on iOS, [.AUTOFILL_HINT_POSTAL_ADDRESS](https://developer.android.com/reference/androidx/autofill/HintConstants#AUTOFILL_HINT_POSTAL_ADDRESS)
+  /// on Android, and [address-line1](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete)
+  /// on web.
+  static const String streetAddressLine1 = 'streetAddressLine1';
+
+  /// The input field expects the second line of a street address.
+  ///
+  /// Translates to [UITextContentType.streetAddressLine2](https://developer.apple.com/documentation/uikit/uitextcontenttype/1649658-streetaddressline2),
+  /// on iOS, [.AUTOFILL_HINT_POSTAL_ADDRESS_EXTENDED_ADDRESS](https://developer.android.com/reference/androidx/autofill/HintConstants#AUTOFILL_HINT_POSTAL_ADDRESS_EXTENDED_ADDRESS)
+  /// on Android, and [address-line2](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete)
+  /// on web.
+  static const String streetAddressLine2 = 'streetAddressLine2';
+
+  /// The input field expects a password.
+  ///
+  /// Translates to [UITextContentType.password](https://developer.apple.com/documentation/uikit/uitextcontenttype/2865813-password),
+  /// on iOS, [.AUTOFILL_HINT_PASSWORD](https://developer.android.com/reference/androidx/autofill/HintConstants#AUTOFILL_HINT_PASSWORD)
+  /// on Android, and [current-password](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete)
+  /// on web.
+  static const String password = 'password';
+
+  /// The input field expects a username or an account name.
+  ///
+  /// Translates to [UITextContentType.username](https://developer.apple.com/documentation/uikit/uitextcontenttype/2866088-username),
+  /// on iOS, [.AUTOFILL_HINT_USERNAME](https://developer.android.com/reference/androidx/autofill/HintConstants#AUTOFILL_HINT_USERNAME)
+  /// on Android, and [username] (https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete)
+  /// on web.
+  static const String username = 'username';
+}
+
+/// A collection of autofill related information that represents an [AutofillClient].
+///
+/// Typically used in [TextInputConfiguration.autofillConfiguration].
+@immutable
+class AutofillConfiguration {
+  /// Creates autofill related configuration information that can be sent to the
+  /// platform.
+  const AutofillConfiguration({
```

Comment:
I remember we discussed making this step easier to use for the users by using different constructors for the common use cases. We can take advantage of named constructors in dart. Some examples might be:

AutofillConfiguration.newPassword(), AutofillConfiguration.password(), AutofillConfiguration.username(), AutofillConfiguration.email(),

Reply:
From
```dart
AutofillConfiguration(
  uniqueIdentifier: "identifier",
  autofillHints: <String>[AutofillHints.password],
  currentEditingValue: editingValue,
);
``` 
To
```dart
AutofillConfiguration.password(
  uniqueIdentifier: "identifier",
  currentEditingValue: editingValue,
);
```
it saves one line and it will only be used by people who subclass `AutofillClient`. I think the default constructor should be good enough for now.

Reply:
Since we will be the author of the constructor, we can write this way:
AutofillConfiguration.password(this.currentEditingValue): this.uniqueIdentifier='currentpswd';

on the other hand this only worths it if the developers need to use it. Then we are providing them with a ready to use configuration. Do you think the developer will be using this themselves or is it only a class which will be used for the framework internals?

Reply:
I think this API will only be useful for developers that implement their own `TextInputClient`. Most developers don't have to instantiate an `AutofillConfiguration` themselves.